### ISSUE TITLE
Out-of-box SSR support

### ISSUE THREAD

**Issue by kachkaev** [getsentry/sentry-replay#246]
👋 folks! Alfa tester here 🧑‍🔬 Curious to hear your thoughts on this one!

### Environment

`"@sentry/replay": "0.6.4-0"`
`"next": "12.3.1"`

### Steps to Reproduce

1. Configure Sentry in a Next.js app [using a wizard](https://docs.sentry.io/platforms/javascript/guides/nextjs/)

2. Reuse one `sentryConfig` in `sentry.client.ts` and `sentry.server.ts` for simplicity (e.g. by importing `sentryConfig` from `sentry.client.ts` to `sentry.server.ts`)

3. Add `new Replay()` to a shared config

4. Try `next dev` or `next build`

### Expected Result

It just works™, i.e. `new Replay()` is a noop on the server

### Actual Result

💥

```
ReferenceError: window is not defined
    at isInternal (/path/to/project/node_modules/@sentry/replay/dist/index.js:158:56)
    at captureInternalException (/path/to/project/node_modules/@sentry/replay/dist/index.js:497:10)
    at new Replay (/path/to/project/node_modules/@sentry/replay/dist/index.js:1219:13)
    at eval (webpack-internal:///./sentry.client.config.ts:18:9)
    at Object../sentry.client.config.ts (/path/to/project/.next/server/pages/_app.js:142:1)
    at __webpack_require__ (/path/to/project/.next/server/webpack-runtime.js:33:42)
    at eval (webpack-internal:///./sentry.server.config.ts:4:79)
    at Object../sentry.server.config.ts (/path/to/project/.next/server/pages/_app.js:1319:1)
    at __webpack_require__ (/path/to/project/.next/server/webpack-runtime.js:33:42)
    at __webpack_exec__ (/path/to/project/.next/server/pages/_app.js:2402:39)
[LATE_SETUP_CALL] {}
```

### Workaround

Do not reuse Sentry config, thus ensuring that `new Replay()` is only called in the client bundle. E.g.:

```ts
 integrations:
    typeof window === "undefined"
      ? []
      : [
          new Replay({
            // options
          }),
        ],
```

I’m fiddling with `sentry-replay` in https://github.com/blockprotocol/blockprotocol/pull/682 – any tips welcome!

**Comment by kachkaev** [comment]
I just upgraded from `0.6.4-0` to `0.6.4` but this did not help. Still getting `ReferenceError: window is not defined` when calling `new Replay()` in `sentry.server.config.ts`

**Comment by ryan953** [comment]
@kachkaev I think we got the wrong commit when we cut the release. Sorry about that.

The release is cut from https://github.com/getsentry/sentry-replay/commit/4eb6879655f875a620ddb21ca60ff27504c21b14

But that commit isn't on the main branch, on [main](https://github.com/getsentry/sentry-replay/commits/main) we've got 77aed39b79533d50c90ad66ceb17a226e70f0376 instead. 

@billyvg Lets bump it again

**Comment by kachkaev** [comment]
Thanks folks! I tried `0.6.5` in https://github.com/blockprotocol/blockprotocol/pull/694 but it did not work. Not sure why, but when I removed `typeof window !== "undefined"` from my Sentry config, API routes broke on Vercel. See PR commits and corresponding CI checks / Vercel deployments. Maybe there is something else with `0.6.5` which breaks lambdas, IDK. I kept `typeof window !== "undefined"` and `0.6.4` for now.

**Comment by kachkaev** [comment]
I tried `0.6.5` in https://github.com/hashintel/hash/pull/1232 and managed to see a proper stack trace on Vercel:

```
RequestId: a50261f7-38f0-4472-9ee8-29e369e66aaa Error: Runtime exited with error: exit status 1
Runtime.ExitError
2022-10-20T18:58:02.838Z	fcbab5bf-9c98-4c29-8173-e6b1c259d1b5	ERROR	ReferenceError: window is not defined
    at isInternal (/var/task/node_modules/@sentry/replay/dist/index.js:158:56)
    at captureInternalException (/var/task/node_modules/@sentry/replay/dist/index.js:497:10)
    at new Replay (/var/task/node_modules/@sentry/replay/dist/index.js:1220:13)
    at Object.6029 (/var/task/packages/hash/frontend/.next/server/pages/api/ory/[...paths].js:28:9)
    at __webpack_require__ (/var/task/packages/hash/frontend/.next/server/webpack-api-runtime.js:25:42)
    at __webpack_exec__ (/var/task/packages/hash/frontend/.next/server/pages/api/ory/[...paths].js:80:39)
    at /var/task/packages/hash/frontend/.next/server/pages/api/ory/[...paths].js:81:28
    at Object.&lt;anonymous&gt; (/var/task/packages/hash/frontend/.next/server/pages/api/ory/[...paths].js:84:3)
    at Module._compile (node:internal/modules/cjs/loader:1105:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1159:10)
RequestId: fcbab5bf-9c98-4c29-8173-e6b1c259d1b5 Error: Runtime exited with error: exit status 1
```

Pointing here:
https://github.com/getsentry/sentry-replay/blob/8b805f26145d04983a1744cc73cd3eaab1f81d20/src/util/isInternal.ts#L5-L7

**Comment by billyvg** [comment]
@ryan953 Ah https://github.com/getsentry/sentry-replay/blob/35903b785ae7346e7d8447190552eeaeb01969d5/src/index.ts#L202-L208 -- that's not covered by `isBrowser`

**Comment by ryan953** [comment]
Yeah damnit. I quickly added a check to cover that spot: https://github.com/getsentry/sentry-replay/pull/253

I'm going to see if we can change the tsconfig and eslint to not allow browser api calls like this, and then we can opt-in each time. That way we'll have a chance to add these guards

**Comment by ryan953** [comment]
https://github.com/hashintel/hash/pull/1232 is using 0.6.6, if that PR merges without issue i'll close this too.

**Comment by kachkaev** [comment]
I tried 0.6.6 in https://github.com/blockprotocol/blockprotocol/pull/698 but had to bring back `typeof window !== "undefined"`.

`yarn build && yarn start` + open `/api/something` gives this:

```
ready - started server on 0.0.0.0:3000, url: http://localhost:3000
info  - Loaded env from /path/to/app/.env.local
Error: Multiple Sentry Session Replay instances are not supported
    at new Replay (/path/to/app/node_modules/@sentry/replay/dist/index.js:1221:25)
    at Object.4074 (/path/to/app/.next/server/chunks/4486.js:282:9)
    at __webpack_require__ (/path/to/app/.next/server/webpack-api-runtime.js:25:42)
    at __webpack_exec__ (/path/to/app/.next/server/pages/api/me.js:215:39)
    at /path/to/app/.next/server/pages/api/me.js:216:94
    at Function.__webpack_require__.X (/path/to/app/.next/server/webpack-api-runtime.js:177:21)
    at /path/to/app/.next/server/pages/api/me.js:216:47
    at Object.<anonymous> (/path/to/app/.next/server/pages/api/me.js:219:3)
    at Module._compile (node:internal/modules/cjs/loader:1155:14)
    at Object.Module._extensions..js (node:internal/modules/cjs/loader:1209:10)
```

**Comment by ryan953** [comment]
I'm glad it's a new error! 

I'll spin up a nextjs app and dig into why it's instantiating the class twice. There could be implications for our regular sdk as well.

**Comment by ryan953** [comment]
So my basic nextjs site seems to work fine with the shared config across `sentry.server.config.js` and `sentry.client.config.js`

I tried checking out git@github.com:blockprotocol/blockprotocol.git and running the app. After setting up everything and running `yarn build && yarn start` I visited `http://localhost:3000/api/me/api-keys` on master branch and saw:
```
{"errors":[{"msg":"You must be logged in to perform this request"}]}
```

Checking out this branch I reproduced the issue. Seems like next.js imports the config all the time, so we're going to work on some possible solutions ranging from 'really naive' to 'too crazy'.
